s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/AWSV2SourceClient.java

```diff
@@ -76,9 +88,50 @@ public AWSV2SourceClient(final S3SourceConfig s3SourceConfig, final Set<String>
         this.s3Client = s3Client;
         this.bucketName = s3SourceConfig.getAwsS3BucketName();
         this.failedObjectKeys = new HashSet<>(failedObjectKeys);
+
+        // TODO the code below should be configured in some sort of taks assignement method/process/call.
```

Comment:
We do have task assignment, and yes we should create a separate PR to tackle this imo. 👍